### DISCUSSION TITLE
How to use the type TooltipContent in own code?

### DISCUSSION THREAD

**Discussion by cg-sat** [visgl/deck.gl#8999] [Q&A]
I'm not an expert in Typescript, so I'm not sure whether I found an issue in the deckgl type declarations or I made a mistake in my code.

I want to show a custom tooltip on a layer and that tooltip should contain HTML. To make the type system happy, I'm using the type `TooltipContent` as the return type for my function. It looks like this: 
```
function getTooltip(info: PickingInfo): TooltipContent {
	return { html: '<b>Some useful text</b>'};
}
```

Unfortunately the type `TooltipContent` isn't exported from `@deck.gl/core`, so I used the following import:
```
import type { TooltipContent } from '@deck.gl/core/lib/tooltip';
```
This import works, but causes a warning in VS Code saying `Cannot find module @deck.gl/core/lib/tooltip or its corresponding type declarations`. Is here a Typescript expert, who can advice on this issue? My impression is, that the type `TooltipContent` should be exported in the file https://github.com/visgl/deck.gl/blob/master/modules/core/src/index.ts , somewhere after line 111.

**Comment by ibgreen** [comment]
FWIW - Typescript has some utility type helpers that can sometimes be used when a type you wanted to use isn't exported: https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype. Perhaps something like that could work in this case?

**Comment by cg-sat** [comment]
Thanks for the suggestion. I tried it, but it doesn't solve the problem, because I still need to import the TooltipContent type from somewhere. I could copy the type from deckgl to my code or use the any type. Both options are not a real solution.

**Comment by galenwinsor** [comment]
Can you share the specific error TS is giving you that makes you need to assert the return type?

**Reply by cg-sat** [reply]
It's a warning in VS Code:
![image](https://github.com/visgl/deck.gl/assets/143688769/9a3fd3f8-4002-4d44-b526-93e110b48e28)

In parallel I opened issue #9013 for this.
